nautobot_golden_config/utilities/helper.py

```diff
@@ -46,21 +47,27 @@ def get_job_filter(data=None):
     elif data.get("device"):
         query.update({"id": data["device"].values_list("pk", flat=True)})
 
-    base_qs = Device.objects.none()
+    raw_qs = Q()
     for obj in models.GoldenConfigSetting.objects.all():
-        base_qs = base_qs | obj.get_queryset().distinct()
+        # If scope is set to {} break out of loop as all devices are in scope.
+        if obj.scope == {}:
+            raw_qs = Q()
+            break
+        raw_qs = raw_qs | obj.dynamic_group.generate_query()
```

Comment:
Suggestion:

Before doing the for loop check to see any `GoldenConfigSetting` has objects that have DG specified:

```py
if not models.GoldenConfigSetting.objects.filter(dynamic_group__isnull=True).exists():
   for obj in models.GoldenConfigSetting.objects.all():
      raw_qs = raw_qs | obj.dynamic_group.generate_query()
```


Saves you loading the details of each GoldenConfigSetting, and potentially getting through a list of a dozen or more settings and the last one having no scope.

Reply:
@bryanculver a DynamicGroup is a required FK https://github.com/nautobot/nautobot-plugin-golden-config/blob/develop/nautobot_golden_config/models.py#L489

Reply:
Good catch I can make that change

Reply:
Edited my comment. Filter gets stored as an empty dictionary always.

Reply:
Just pushed `dynamic_group__filter__iexact="{}"`, should I do `dynamic_group__filter={}`

Reply:
> Just pushed `dynamic_group__filter__iexact="{}"`, should I do `dynamic_group__filter={}`

Good enough for now. I think the only thing that would make the latter more stable is if the field type changed or how Django implements `JSONField`s.